api/client/service/opts.go

```diff
@@ -545,6 +548,7 @@ const (
 	flagRestartWindow      = "restart-window"
 	flagStopGracePeriod    = "stop-grace-period"
 	flagUpdateDelay        = "update-delay"
+	flagUpdateOnFailure    = "update-on-failure"
```

Comment:
As I mentioned in swarmkit, I feel like this flag name is really really ambiguous.
`update-failure-action`, `on-update-failure`, `update-failure-policy`